jetty-io/src/main/java/org/eclipse/jetty/io/ClientConnector.java

```diff
@@ -221,20 +238,16 @@ public void connect(SocketAddress address, Map<String, Object> context)
             context.put(ClientConnector.CLIENT_CONNECTOR_CONTEXT_KEY, this);
             context.putIfAbsent(REMOTE_SOCKET_ADDRESS_CONTEXT_KEY, address);
 
-            channel = SocketChannel.open();
+            SocketChannelWithAddress channelWithAddress = factory.newSocketChannelWithAddress(address, context);
+            channel = channelWithAddress.getSocketChannel();
+            address = channelWithAddress.getSocketAddress();
             SocketAddress bindAddress = getBindAddress();
             if (bindAddress != null)
-            {
-                boolean reuseAddress = getReuseAddress();
-                if (LOG.isDebugEnabled())
-                    LOG.debug("Binding to {} to connect to {}{}", bindAddress, address, (reuseAddress ? " reusing address" : ""));
-                channel.setOption(StandardSocketOptions.SO_REUSEADDR, reuseAddress);
-                channel.bind(bindAddress);
-            }
+                bind(channel, bindAddress);
             configure(channel);
 
             boolean connected = true;
-            boolean blocking = isConnectBlocking();
+            boolean blocking = isConnectBlocking() && address instanceof InetSocketAddress;
```

Comment:
Because in the code below I cannot call `channel.socket()` if it is a Unix-Domain socket address (it throws).
I would do `channel instanceof SocketChannel`, but that is true for both network and Unix-Domain, despite the fact that calling `socket()` on a `SocketChannel` may throw if it is Unix-Domain. 🤷🏼‍♂️